### PR TITLE
Update to PhpStan v1

### PR DESCRIPTION
**Pull request by blankse**
Update PhpStan to the new 1.0.0 version:
https://github.com/phpstan/phpstan/releases/tag/1.0.0

### CODE REVIEW

File: models/DataObject/Concrete.php [hunk 1]
```diff
@@ -663,7 +663,7 @@ public static function __callStatic($method, $arguments)
 
             $list = static::makeList($listConfig, $objectTypes);
 
-            if ($field instanceof AbstractRelations && $field->isFilterable()) {
```
Comment (dvesh3) [inline models/DataObject/Concrete.php:666]: why removed the `isFilterable()` check?

File: models/DataObject/Localizedfield/Dao.php [hunk 2]
```diff
@@ -849,7 +849,7 @@ public function createUpdateTable($params = [])
                                 $this->addModifyColumn($table, $key . '__' . $fkey, $fvalue, '', 'NULL');
                                 $protectedColumns[] = $key . '__' . $fkey;
                             }
-                        } elseif ($value->getColumnType()) {
```
Comment (dvesh3) [inline models/DataObject/Localizedfield/Dao.php:856]: return value can be null

File: models/DataObject/Concrete.php (hunk 1 above)
Comment (blankse) [reply models/DataObject/Concrete.php:666]: See https://github.com/pimcore/pimcore/blob/9c2c366e7b96b5994828bb8c33d5561737cb7383/models/DataObject/Concrete.php#L610-L612

File: models/DataObject/Localizedfield/Dao.php (hunk 2 above)
Comment (blankse) [reply models/DataObject/Localizedfield/Dao.php:856]: See https://github.com/pimcore/pimcore/blob/9c2c366e7b96b5994828bb8c33d5561737cb7383/models/DataObject/Localizedfield/Dao.php#L843

**Review by dvesh3 — APPROVED** [review]

**Comment by blankse** [comment]
I think it would be good to merge https://github.com/pimcore/pimcore/pull/10964 first, there the deprecations are better fixed.

I can rebase/merge this here afterwards.

File: lib/Targeting/ActionHandler/AssignTargetGroup.php [hunk 3]
```diff
@@ -19,29 +19,21 @@
...
-     */
-    private $conditionMatcher;
-
     /**
      * @var TargetingStorageInterface
      */
     private $storage;
 
     public function __construct(
-        ConditionMatcherInterface $conditionMatcher,
```
Comment (dvesh3) [inline lib/Targeting/ActionHandler/AssignTargetGroup.php:41]: @blankse CMF extends this class and throws exception  after this change:
image
could you take care of that on CMF bundle? thanks!

Comment (blankse) [reply lib/Targeting/ActionHandler/AssignTargetGroup.php:41]: @dvesh3 I added it again with comment. Created an PR for CMF (https://github.com/pimcore/customer-data-framework/pull/247) so it can removed in Pimcore 11.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,14 @@
+/bin/console
```
Comment (dvesh3) [inline .gitignore]: Is this change related? If not, then please revert and put it in a separate PR. thanks!

Comment (blankse) [reply .gitignore]: If you setup the test environment these files/folders are not ignored. Yes I can split it to an other PR if you like.

Comment (dvesh3) [reply .gitignore]: yes, please 😊

Comment (blankse) [reply .gitignore]: @dvesh3 Done

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@blankse thanks a lot!
